presidio-api/cmd/presidio-api/api/scanner-cron-job/scanner-cron-job.go

```diff
@@ -0,0 +1,82 @@
+package scannercronjob
+
+import (
+	"context"
+	"fmt"
+
+	types "github.com/Microsoft/presidio-genproto/golang"
+	"github.com/Microsoft/presidio/pkg/presidio"
+	store "github.com/Microsoft/presidio/presidio-api/cmd/presidio-api/api"
+	"github.com/Microsoft/presidio/presidio-api/cmd/presidio-api/api/templates"
+)
+
+//ScheduleScannerCronJob schedule scanner cron job
+func ScheduleScannerCronJob(ctx context.Context, api *store.API, cronAPIJobRequest *types.ScannerCronJobApiRequest, project string) (*types.ScannerCronJobResponse, error) {
+
+	scannerCronJobRequest, err := getScannerCronJobRequest(api, cronAPIJobRequest, project)
+	if err != nil {
+		return nil, err
+	}
+	schedulerResponse, err := invokeScannerCronJobScheduler(ctx, api.Services, scannerCronJobRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	return schedulerResponse, nil
+
+}
+
+func invokeScannerCronJobScheduler(ctx context.Context, services presidio.ServicesAPI, scannerCronJobRequest *types.ScannerCronJobRequest) (*types.ScannerCronJobResponse, error) {
+
+	res, err := services.ApplyScan(ctx, scannerCronJobRequest)
+	if err != nil {
+		return nil, err
+	}
+	return res, nil
+}
+
+func getScannerCronJobRequest(api *store.API, cronJobAPIRequest *types.ScannerCronJobApiRequest, project string) (*types.ScannerCronJobRequest, error) {
+	scanRequest := &types.ScanRequest{}
+	trigger := &types.Trigger{}
+	var name string
+
+	if cronJobAPIRequest.ScannerCronJobTemplateId != "" {
```

Comment:
you should add validate function same as in the anonymizer..

Reply:
Will do in a separate PR for scanner and streams with other features